### ISSUE TITLE
Add optional `screen_id` param to `get_display_cutouts()` and `get_display_safe_area()`

### ISSUE THREAD

**Issue by AdriaandeJongh** [godotengine/godot-proposals#10647]
### Describe the project you are working on

[Rift Riff](https://store.steampowered.com/app/2800900/Rift_Riff/) and soon a sequel to [Hidden Folks](https://hiddenfolks.com).

### Describe the problem or limitation you are having in your project

As of the last few years, MacBook screens (and maybe other laptops?) have cutouts at the top of the screen. `get_display_cutouts()` and `get_display_safe_area()` works as expected if the user only has one screen connected, but it's ambiguous when the user has an external screen connected.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add an optional `screen_id` param to `get_display_cutouts()` and `get_display_safe_area()` so that you can get the display cutouts and safe area of a _specific_ screen as supposed to _the_ (?) screen.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I don't know anything about the inner workings of the engine so I am of no help here. I am not aware of any external screens with safe areas or cutouts, so perhaps the implementation is simple: only return the safe area and cutouts for screens that are not external – whatever that means...?

### If this enhancement will not be used often, can it be worked around with a few lines of script?

n/a

### Is there a reason why this should be core and not an add-on in the asset library?

n/a

**Comment by AThousandShips** [comment]
How does MacBook relate to this? This is an Android specific thing?

**Comment by AdriaandeJongh** [comment]
It is not an Android specific thing. All recent Apple devices (iPhones, MacBooks, Apple TVs, etc) return safe areas other than their full screen size. Calling `get_display_safe_area()` on a MacBook rightfully returns the safe area – but like I said, it is ambiguous as to what screen that rect applies to.

Aside from that, I do believe Android can theoretically have external monitors, in which case this applies on Android as well.

**Comment by AThousandShips** [comment]
Yes but *Godot* doesn't support cutouts on any other device, so that would first have to be implemented on macOS

The `get_display_safe_area` works by "default", as it just removes all the zero cutouts from `get_display_cutouts`, as per the documentation:
> Currently only implemented on Android. Other platforms will return an empty array even if they do have display cutouts or notches.

**Comment by AdriaandeJongh** [comment]
Hmm. `get_display_safe_area()` works on iOS as well, so it seems it is already implemented on multiple platforms.

**Comment by AThousandShips** [comment]
Yes, it is, but it just gets the area and removes the cutouts, aka none, or *it* works but `get_display_cutouts` is not

**Comment by AdriaandeJongh** [comment]
You're right. I guess the point I'm trying to make here is: it makes sense for both of these functions to have screen_id parameters, because both cutouts and safe areas are dependent on that.

**Comment by AThousandShips** [comment]
Would be good to get some information on what platforms would support such an argument, if the various APIs support that and how

**Comment by geekley** [comment]
I've opened a more generic proposal for it to be added to Window, possibly with a new built-in node to go with it, so you don't have to worry about screen id or window positioning, and it should also push away from rounded corners: #11106
